core/src/main/scala/org/bitcoins/core/util/Bech32.scala

```diff
@@ -248,3 +311,22 @@ object Bech32 extends Bech32 {
     'g', 'f', '2', 't', 'v', 'd', 'w', '0', 's', '3', 'j', 'n', '5', '4', 'k',
     'h', 'c', 'e', '6', 'm', 'u', 'a', '7', 'l')
 }
+
+abstract class Bech32HumanReadablePart {
+  require(chars.forall(Bech32.isInHrpRange),
+          s"Some characters in $chars were not in valid HRP range ([33-126])")
+
+  def chars: String
+
+  /** Expands this HRP into a vector of UInt5s, in accordance with the Bech32 spec */
+  def expand: Vector[UInt5] = Bech32.hrpExpand(this)
+}
+
+object Bech32HumanReadablePart {
```

Comment:
I'm against this, it doesn't make sense since there isn't a "concrete" Bech32HumanReadablePart. This needs to be an abstract type, and then we pattern match during the parsing phase to figure what network it is associated with

Reply:
I went down this road today, but realized this breaks tests from Sipa: https://github.com/sipa/bech32/blob/master/ref/python/tests.py#L34

It is possible for a `Bech32HumanReadablePart` to exist independently, as the standard isn't necessarily tied to Bitcoin